monitor/monitor.go

```diff
@@ -29,14 +28,35 @@ import (
 	"github.com/cilium/cilium/pkg/lock"
 )
 
-const pollTimeout = 5000
+const (
+	pollTimeout = 5000
+
+	// queueSize is the size of the message queue
+	queueSize = 65536
+)
 
 var (
 	mutex         lock.Mutex
-	listeners     = list.New()
+	listeners     = make(map[*monitorListener]struct{})
```

Comment:
Can't you use `map[monitorListener]chan []byte` instead?

Reply:
What's the advantage of that? It's not extendable.

Reply:
I was trying to solve the problem of using a pointer as a key of a map. That can give problems when doing something like

```
func Foo(m monitorListener){
   _, exists := listeners[&m]
   // exists will always be false
}

```

Reply:
Sure, because it's a different address.